### ISSUE TITLE
uri has been modified

### ISSUE THREAD

**Issue by YaYaB** [jolibrain/deepdetect#740]
(A Markdown syntax reminder is available here: https://guides.github.com/features/mastering-markdown/)

#### Configuration
- Version of DeepDetect:
    - [ X] Locally compiled on:
        - [X] Ubuntu 16.04 LTS
        - [ ] Mac OSX
        - [ ] Other:
    - [ ] Docker
    - [ ] Amazon AMI
- Commit (shown by the server when starting):
c6d27f8d69336b89eb1467e44ffe1d2cd350bd87

#### Your question / the problem you're facing:
I am using a training an image model with local paths to images. When I try to make the predictions on the lmdbs I get the uris of the images modified. A prefix is added before as an index, this is OK and expected. However the end of the uri has been cut.

After a look at the code I saw that in [here](https://github.com/jolibrain/deepdetect/blob/a68b6c26ee92dbde047d240d3e1506c9732b681a/src/backends/caffe/caffeinputconns.cc#L370), the uri's length is limited to 256. Unfortunately because of this, it is complicated to map the uri to the real image behind.

I see two possible solutions:
- Indicate in the README that ur

**Comment by beniz** [comment]
Ah hi @YaYaB good catch, as usual! So one way we have here is that we only store an id, or the relevant part of the URI, this is why we accomodate the 256 chars limit in our applications.

One of the reasons for using an internal ID to your application is that there can't be any good max size that'd fit for all cases, obviously.

Agreed that this should be part of the doc / readme, where do you believe it's most useful we add a warning ?

**Comment by YaYaB** [comment]
Being part of doc / readme would be amazing. And also having a warning when the lmdb is created could be nice as well

**Comment by beniz** [comment]
OK, regarding the warning, we may allow it only once, because otherwise, it'd get printed millions of times for large datasets for which most of the time, the id in the db doesn't matter.

**Comment by YaYaB** [comment]
Yeah sure, once whem the lmdb has finished or something like this

**Comment by beniz** [comment]
See if #755 is fine for your need.

**Comment by YaYaB** [comment]
Yep it's perfect, it would be even better to have a note on the Readme or the api doc :)
